autovalet_goal_generation/scripts/transfer_lane.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+'''
+Script runs an approximate time policy message filter to catch
+depthmap and segmentation maps at the same time and produces a
+segmented lane cloud and publishes it
+
+Author  : Subramanian Krishnan (subramak@andrew.cmu.edu)
+Date    : 09 September 2020
+
+Changelog:
+    subbu - 9/07 - Initial commit
+    subbu - 9/07 - Lane seg overlaid on depthmap
+    eva/subbu - 9/12 - Fixed fov
+    subbu - 9/14 - lane line extraction from segmap
+    subbu - 9/18 - Depth registration added
+'''
+import numpy as np
+import rospy
+import sensor_msgs.point_cloud2 as pcl2
+import cv2
+
+from sensor_msgs.msg import Image, CameraInfo, PointCloud2
+from std_msgs.msg import Header
+from cv_bridge import CvBridge, CvBridgeError
+from message_filters import ApproximateTimeSynchronizer, Subscriber
+
+class transfer:
+
+  def __init__(self, segmap_topic, depthmap_topic, cameraInfo_topic, laneCloud_topic):
+    self.bridge = CvBridge()
+    self.depth_sub  = Subscriber(depthmap_topic, Image)
+    self.segmap_sub = Subscriber(segmap_topic, Image)
+
+    # Read camera info by looking up only one message
+    self.camera = rospy.wait_for_message(cameraInfo_topic, CameraInfo)
+    self.lane_cloud_pub = rospy.Publisher(laneCloud_topic, PointCloud2, queue_size=1)
+
+    # Looking to synchronize both the topics within a 1/10th of a second
+    self.ats = ApproximateTimeSynchronizer([self.depth_sub, self.segmap_sub], queue_size=2, slop=0.1)
+    self.ats.registerCallback(self.callback)
+
+    # Define the camera matrix individual values for projection
+    self.fx     = self.camera.K[0]
```

Comment:
where's the 5th dof of K?

Reply:
I believe there are only 4: cx, cy, fx, fy

Reply:
skew is assumed zero